e2e/nomostest/nt.go

```diff
@@ -793,59 +788,69 @@ func (nt *NT) MustDeleteGatekeeperTestData(file, name string) {
 	nt.MustKubectl("delete", "-f", absPath, "--ignore-not-found", "--wait")
 }
 
-// PortForwardOtelCollector forwards the otel-collector pod.
-func (nt *NT) PortForwardOtelCollector() {
-	// Retry otel-collector port-forwarding in case it is in the process of upgrade.
-	took, err := Retry(60*time.Second, func() error {
-		pod, err := nt.GetDeploymentPod(ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace)
-		if err != nil {
-			return err
-		}
-		// The otel-collector forwarding port needs to be updated after otel-collector restarts or starts for the first time.
-		// It sets otelCollectorPodName and otelCollectorPort to point to the current running pod that forwards the port.
-		if pod.Name != nt.otelCollectorPodName {
-			port, err := nt.ForwardToFreePort(ocmetrics.MonitoringNamespace, pod.Name, testmetrics.MetricsPort)
-			if err != nil {
-				return err
-			}
-			nt.otelCollectorPort = port
-			nt.otelCollectorPodName = pod.Name
+// portForwardOtelCollector forwards the otel-collector pod.
+func (nt *NT) portForwardOtelCollector() {
+	if nt.otelCollectorPort == 0 {
+		var mux sync.Mutex
```

Comment:
Not sure i like having the lock outside of the struct that owns the protected field.
Won't this replace the lock if the value's set to zero?

I suggest moving this all into a PortForwarder object and having one instance each for otel-collector and git-server.

Reply:
This shouldn't happen because the value should never get set back to zero, but I suppose a struct pattern might be a better fit. I'll take a pass at refactoring this

Reply:
Updated to use a `portForward` struct. PTAL